### ISSUE TITLE
make a benchmark configuration for MLDG

### ISSUE THREAD

**Issue by smilesun** [marrlab/DomainLab#172]
Li, Da, et al. “Learning to generalize: Meta-learning for domain generalization.” Proceedings of the AAAI conference on artificial intelligence. Vol. 32. No. 1. 2018.

Estimated amounts of work: 3hours

**Comment by Car-la-F** [comment]
@smilesun,
Reading the paper I found 3 Hyperparameters coming with the optimizer MLDG:

- $\alpha$: learning rate for gradient descend in the meta learning process (meta-train step size)
- $\beta$: weights meta-train and meta-test (loss = "meta train" + $\beta$ "meta test")
- $\gamma$: learning rate of the meta-optimization

I tried to understand how these hyperparameters are set in domainlab and made the observations below. @smilesun can you please comment on them for me to understand which hyperparameter shall be respected in the benchmark.

---

$\alpha$:
the meta learning (just using the meta training samples) uses the function 

https://github.com/marrlab/DomainLab/blob/23122c87aa36a7566c15ab3dffe3997d8b9f8c37/domainlab/algos/trainers/train_basic.py#L43

where the learning rate specified in `--lr` is used

https://github.com/marrlab/DomainLab/blob/23122c87aa36a7566c15ab3dffe3997d8b9f8c37/domainlab/algos/trainers/train_basic.py#L47

---

$\beta$:
From my understanding the following line of code computes the loss function $F(\Theta) + \beta G(\Theta - \alpha F'(\Theta))$ from the paper.
`loss_source.sum()` = $F(\Theta) $
`loss_look_foreward.sum()` = $G(\Theta - \alpha F'(\Theta))$

https://github.com/marrlab/DomainLab/blob/23122c87aa36a7566c15ab3dffe3997d8b9f8c37/domainlab/algos/trainers/train_mldg.py#L86

**At which point does $\beta$ come into the loss function? Is $\beta$ already included in loss_look_foreward.sum() ?**
For me it does make sense that `loss_look_foreward.sum()` can be seen as a regularisation term, hence one should be able to set $\beta$ using `--gamma_reg`. Is this the case? 

---

$\gamma$ is the learning rate of the meta-optimizer, which has learning rate specified in the comand line arguments with `--lr`

https://github.com/marrlab/DomainLab/blob/23122c87aa36a7566c15ab3dffe3997d8b9f8c37/domainlab/algos/trainers/train_mldg.py#L21

**Comment by smilesun** [comment]
Thanks, I have added gamma_reg in this commit: 
[214f667](https://github.com/marrlab/DomainLab/commit/214f667fc005ea3a4431b194b9618fecbb9952f3)

**Comment by smilesun** [comment]
i think we can set the same learning rate for meta and the major optimizer
